pymongo/collection.py

```diff
@@ -356,8 +380,13 @@ def with_options(self, codec_options=None, read_preference=None,
                           write_concern or self.write_concern,
                           read_concern or self.read_concern)
 
-    def bulk_write(self, requests, ordered=True,
-                   bypass_document_validation=False, session=None):
+    def bulk_write(
+        self,
+        requests: List[WriteOp],
```

Comment:
Given a file
```py
collection = Collection()  # type: ignore
requests = [InsertOne({})]
collection.bulk_write(requests)
```

running mypy gives us
```
error: Argument 1 to "bulk_write" of "Collection" has incompatible type "List[InsertOne]"; expected "List[Union[InsertOne, DeleteOne, DeleteMany, ReplaceOne, UpdateOne, UpdateMany]]"
note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
note: Consider using "Sequence" instead, which is covariant
```

Changing this line to
```py
        requests: Sequence[WriteOp],
```
resolves this issue.

`common.validate_list("requests", requests)` on [line R455](https://github.com/mongodb/mongo-python-driver/pull/829/files#diff-00c0d0047a08351562fe8a665b8530f6fc909d6634389104a8982e01c190feb5R455) would likely best be replaced by sequence validation instead of list validation.

(This issue also appears in [mongodb-labs/pymongo-stubs](https://github.com/mongodb-labs/pymongo-stubs/blob/d9d1b0d00f54d179ee81174aeb371356776fbe3a/pymongo-stubs/collection.pyi#L67).)

Reply:
Updated to use `Sequence[WriteOp]` and allow tuples as inputs in several places in eb6bf52